### PR TITLE
WIP: Unity rpc fixes

### PR DESCRIPTION
**Pull request by morpheyesh**

### CODE REVIEW

File: Sources/environments/RpcCommunicator.swift [hunk 1]
```diff
@@ -35,16 +35,33 @@ class UnityToExternalServicerImplementation : CommunicatorObjects_UnityToExterna
...
     
-    func initialize(request: CommunicatorObjects_UnityMessageProto, context: StatusOnlyCallContext) -> EventLoopFuture<CommunicatorObjects_UnityMessageProto> {
+    func initialize(request: CommunicatorObjects_UnityMessageProto, context: StatusOnlyCallContext) ->
+    EventLoopFuture<CommunicatorObjects_UnityMessageProto> {
         return handle(context, request)
     }
     
     func exchange(request: CommunicatorObjects_UnityMessageProto, context: StatusOnlyCallContext) -> EventLoopFuture<CommunicatorObjects_UnityMessageProto> {
+        print("========?>>>>>")
+     var rlIn = CommunicatorObjects_UnityRLInputProto()
```
Comment (SercanKaraoglu) [inline Sources/environments/RpcCommunicator.swift:45]: revert this part

File: Sources/environments/RpcCommunicator.swift [hunk 2]
```diff
@@ -35,16 +35,33 @@ class UnityToExternalServicerImplementation : CommunicatorObjects_UnityToExterna
...
+         rlIn.command = CommunicatorObjects_CommandProto.quit
+         rlIn.sideChannel = request.unityOutput.rlOutput.sideChannel
+         var value = CommunicatorObjects_UnityInputProto()
+         value.rlInput = rlIn
+         var ret = CommunicatorObjects_UnityMessageProto()
+         ret.unityInput = value
+        print(value)
+         return context.eventLoop.makeSucceededFuture(ret)
+    /*EventLoopFuture<CommunicatorObjects_UnityMessageProto> {
+        print("exchange -->>>")
```
Comment (SercanKaraoglu) [inline Sources/environments/RpcCommunicator.swift:55]: this needs to be uncommented instead

**Review by SercanKaraoglu — CHANGES_REQUESTED** [review]

File: Sources/environments/RpcCommunicator.swift (hunk 1 above)
Comment (morpheyesh) [reply Sources/environments/RpcCommunicator.swift:45]: Indeed. I will clean it up as its still WIP :)

Comment (SercanKaraoglu) [reply Sources/environments/RpcCommunicator.swift]: or try this instead :)

```swift
    fileprivate func handle(_ context: StatusOnlyCallContext, _ request: CommunicatorObjects_UnityMessageProto) -> EventLoopFuture<CommunicatorObjects_UnityMessageProto> {
        var rlIn = CommunicatorObjects_UnityRLInputProto()
        rlIn.command = CommunicatorObjects_CommandProto.reset
        rlIn.sideChannel = request.unityOutput.rlOutput.sideChannel
        var response = CommunicatorObjects_UnityMessageProto()
        response.header.status = 200
        response.unityInput = CommunicatorObjects_UnityInputProto()
        response.unityInput.rlInput = rlIn
        response.unityInput.rlInitializationInput = CommunicatorObjects_UnityRLInitializationInputProto()
        response.unityInput.rlInitializationInput.capabilities.baseRlcapabilities = true
        response.unityInput.rlInitializationInput.packageVersion = "1.0.4"
        response.unityInput.rlInitializationInput.communicationVersion = "1.0.0"
        print("request")
        print(request)
        print("response")
        print(response)
        return context.eventLoop.makeSucceededFuture(response)
    }
    func initialize(request: CommunicatorObjects_UnityMessageProto, context: StatusOnlyCallContext) -> EventLoopFuture<CommunicatorObjects_UnityMessageProto> {
        return handle(context, request)
    }
    func exchange(request: CommunicatorObjects_UnityMessageProto, context: StatusOnlyCallContext) -> EventLoopFuture<CommunicatorObjects_UnityMessageProto> {
        return handle(context, request)
    }
```

**Pull request closed without merging**